sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_access_control_client.py

```diff
@@ -113,48 +112,51 @@ def list_role_assignments(self, role_scope, **kwargs):
         )
 
     @distributed_trace
-    def set_role_definition(self, role_scope, permissions, **kwargs):
-        # type: (Union[str, KeyVaultRoleScope], Iterable[KeyVaultPermission], **Any) -> KeyVaultRoleDefinition
+    def set_role_definition(self, role_scope, role_definition_name=None, **kwargs):
+        # type: (Union[str, KeyVaultRoleScope], Optional[Union[str, UUID]], **Any) -> KeyVaultRoleDefinition
         """Creates or updates a custom role definition.
 
         :param role_scope: scope of the role definition. :class:`KeyVaultRoleScope` defines common broad scopes.
             Specify a narrower scope as a string. Managed HSM only supports '/', or KeyVaultRoleScope.GLOBAL.
         :type role_scope: str or KeyVaultRoleScope
-        :param permissions: the role definition's permissions. An empty list results in a role definition with no action
-            permissions.
-        :type permissions: Iterable[KeyVaultPermission]
-        :keyword str role_name: the role's name. If unspecified when creating or updating a role definition, the role
-            name will be set to an empty string.
-        :keyword role_definition_name: the role definition's name. Must be a UUID.
+        :param role_definition_name: the unique role definition name. Unless a UUID is provided, a new role definition
+            will be created with a generated unique name. Providing the unique name of an existing role definition will
+            update that role definition.
         :type role_definition_name: str or uuid.UUID
+        :keyword str role_name: the role's display name. If unspecified when creating or updating a role definition, the
+            role name will be set to an empty string.
         :keyword str description: a description of the role definition. If unspecified when creating or updating a role
             definition, the description will be set to an empty string.
+        :keyword permissions: the role definition's permissions. If unspecified when creating or updating a role
+            definition, the role definition will have no action permissions.
+        :type permissions: Iterable[KeyVaultPermission]
```

Comment:
When I render docs locally, I see `type` doesn't work with `keyword` directives:
```suggestion
        :paramtype permissions: Iterable[KeyVaultPermission]
```